cmake/Modules/UseThreads.cmake

```diff
@@ -0,0 +1,49 @@
+# - Use PThread features
+#
+# Synopsis:
+#
+#	find_threads (module)
+#
+# where:
+#
+#	module            Name of the module to which Threads support
+#	                  etc. should be added, e.g. "opm-core".
+#
+# Note: Compiler flags are always added globally, to avoid ABI
+# incompatibility problems.
+#
+# It is assumed that the following variables are available
+#
+#	${module}_QUIET      Verbosity level of the parent's find module
+#	${module}_LIBRARIES  List of libraries to which Thread support will be added
+#
+# Example:
+#	find_threads (opm-core)
+
+include (AddOptions)
+include (UseCompVer)
+is_compiler_gcc_compatible ()
+
+macro(find_threads opm)
+# default is that Threads are added
+option(USE_PTHREAD CACHE ON)
```

Comment:
CACHE means 'write value even if it is already in cache'. whatever value the user might pre-supply will simply be ignored.